zokrates_analysis/src/flatten_complex_types.rs

```diff
@@ -481,7 +481,7 @@ impl<'ast, T: Field> Flattener<T> {
 // }
 #[derive(Default)]
 pub struct ArgumentFinder<'ast, T> {
-    pub identifiers: HashMap<zir::Identifier<'ast>, zir::Type>,
+    pub identifiers: BTreeMap<zir::Identifier<'ast>, zir::Type>,
```

Comment:
Changed this to `BTreeMap` because it turns out the order is important in some cases